[docs] add troubleshooting abctl page + move trb. from  quickstarts (#45460)
airbytehq/airbyte: docs/deploying-airbyte/troubleshoot-deploy.md

# Troubleshooting abctl

> In this page you're going to find common errors users reported trying to use `abctl`.

Reviewer: ```suggestion
This guide will help you navigate any issues with deploying. This guide is intended for users of `abctl`.
```

Additionally, verify that you can run Docker containers in general by starting with `docker run hello-world`.

```shell
 unable to create kind cluster: failed to init node with kubeadm: 
 command "docker exec --privileged airbyte-abctl-control-plane kubeadm init
  --skip-phases=preflight --config=/kind/kubeadm.conf --skip-token-print --v=6" 
  failed with error: exit status 1
 ```

___

### Time Out Waiting for the Condition

- Error: `timed out waiting for the condition`

```shell
unable to install airbyte chart: unable to install helm: failed pre-install:
 1 error occurred: * timed out waiting for the condition
```

___

### Not able to ingress direct IP addresses

- Error: `must be a DNS name, not an IP address`
- Github Issue: [#110](https://github.com/airbytehq/abctl/pull/110)
- Status: **Solved**

:::tip
Upgrade to `abctl` version `>=0.15.0` default support exposing IP.
:::

```shell
unable to create ingress: Ingress.networking.k8s.io "ingress-abctl" is invalid: 
spec.rules[2].host: Invalid value: "0.0.0.0": must be a DNS name, not an IP address
```

___

### Unable to Read Values from YAML file

- Error: `unable to read values from yaml file`

:::tip

> Check if you're in the right directory or informing the path to the file correctly.

Reviewer: ```suggestion
Verify that you are in the correct directory or informing the path to the file correctly.
```

:::

```shell
unable to merge values with values file './values.yaml': unable to read values 
from yaml file './values.yaml': failed to read file ./values.yaml: open ./values.yaml:
 no such file or directory
```

___

### Could not Create REST Config

- Error: `could not create rest config`

> This error can happen in cases `abctl` doesn't have permission to create file and folders in your system.

Reviewer: ```suggestion
This error can happen when `abctl` does not have permission to create file and folders in your system.
```

```shell
unable to initialize local command: error communicating with kubernetes: 
could not create rest config: stat /root/.airbyte/abctl/abctl.kubeconfig: 
no such file or directory
```

___

### Failed to Create Patch Order in Patch List

- Error: `failed to create patch: The order in patch list`
- Github [#114](https://github.com/airbytehq/abctl/pull/114)
- Status: Fixing

> This error happens sometimes when user had installed `abctl local install` and after run `abctl local install --low-resource-mode`.

Reviewer: ```suggestion
This error can occur when installing using `abctl local install` and subsequently ran `abctl local install --low-resource-mode`.
```

Airbyte is aware of situations where the email is not be automatically populated and we are working on addressing this within the `abctl` tool.  In the interim, some manually steps are required to retrieve the authentication email address when it is unknown.
:::

If the email address for authenticating is not automatically populated, you can set an email with the following command:
```
abctl local credentials --email <USER@COMPANY.EXAMPLE>
```

The password for this user can be retrieved by running `abctl local credentials`.

### Using Custom Connectors

In order to run a custom connector with an Airbyte instance that is running in kind, you must load the docker image of that connector into the cluster. A connector container can be loaded using the following command:
```
kind load docker-image <image-name>:<image-tag> -n airbyte-abctl
```

For more troubleshooting information review the troubleshooting section in [Uploading Customer Connectors](../../operator-guides/using-custom-connectors#troubleshooting)

## Additional Resources

There are several channels for community support of local setup and deployment. 

> **GitHub Airbyte Forum's Getting Started FAQ:**<br/>Search the questions others have asked or ask a new question of your own in the [GitHub forum](https://github.com/airbytehq/airbyte/discussions/categories/questions).

Reviewer: Why does this have br tags? Was it copied over?
I don't think they are needed with markdown

Author: I moved from Quickstarts as is.

**Community Slack:**<br/>Helpful channels for troubleshooting include:<br/>
- [#ask-community-for-troubleshooting](https://airbytehq.slack.com/archives/C021JANJ6TY): Where members of the Airbyte community can ask and answer questions. 
- [#ask-ai](https://airbytehq.slack.com/archives/C01AHCD885S): For quick answers sourced from documentation and open support channels, you can have a chat with our virtual Airbyte assistant.  

**Introductory Course:**<br/>
On Udemy, [The Complete Hands-on Introduction to Airbyte](https://www.udemy.com/course/the-complete-hands-on-introduction-to-airbyte/) is a convenient and hands-on introduction to Airbyte that includes setting up example source and destination configurations. You'll also go on to use it in conjunction with Apache Airflow, Snowflake, dbt, and more.

**Bug Reports:**<br/>If you find an issue with the `abctl` command, please report it as a github
issue [here](https://github.com/airbytehq/airbyte/issues) with the type of `🐛 [abctl] Report an issue with the abctl tool`.

> **Releases:**<br/>If you'd like to select which release of abctl to run, you can find the list of releases [here](https://github.com/airbytehq/abctl/releases/).

Reviewer: ```suggestion
**Releases:** To select a prior release of `abctl` to run, you can find the list of releases [here](https://github.com/airbytehq/abctl/releases/).
```
